src/OSVRTrackedController.h

```diff
@@ -0,0 +1,155 @@
+    friend class ServerDriver_OSVR;
+
+public:
+    OSVRTrackedController(osvr::clientkit::ClientContext& context, int controller_index);
+    //OSVRTrackedController(osvr::clientkit::ClientContext& context);
+
+    virtual ~OSVRTrackedController();
+
+    // ------------------------------------
+    // Management Methods
+    // ------------------------------------
+
+    /**
+     * This is called before an HMD is returned to the application. It will
+     * always be called before any display or tracking methods. Memory and
+     * processor use by the ITrackedDeviceServerDriver object should be kept to
+     * a minimum until it is activated.  The pose listener is guaranteed to be
+     * valid until Deactivate is called, but should not be used after that
+     * point.
+     */
+    virtual vr::EVRInitError Activate(uint32_t object_id) OSVR_OVERRIDE;
+
+    /**
+     * This is called when The VR system is switching from this Hmd being the
+     * active display to another Hmd being the active display. The driver should
+     * clean whatever memory and thread use it can when it is deactivated.
+     */
+    virtual void Deactivate() OSVR_OVERRIDE;
+
+    // ------------------------------------
+    // Controller Methods
+    // ------------------------------------
+
+    /**
+     * Gets the current state of a controller.
+     */
+    virtual vr::VRControllerState_t GetControllerState() OSVR_OVERRIDE;
+
+    /**
+     * Returns a uint64 property. If the property is not available this function will return 0.
+     */
+    virtual bool TriggerHapticPulse(uint32_t axis_id, uint16_t pulse_duration_microseconds) OSVR_OVERRIDE;
+
+protected:
+    const char* GetId();
+
+private:
+    void configure();
+    void configureProperties();
+
+    void freeInterfaces();
+
+    /**
+     * Callback function which is called whenever new data has been received
+     * from the tracker.
+     */
+    static void controllerTrackerCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_PoseReport* report);
+    static void controllerButtonCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_ButtonReport* report);
+    static void controllerButtonTouchCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_ButtonReport* report);
+    static void controllerTriggerCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_AnalogReport* report);
+    static void controllerJoystickXCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_AnalogReport* report);
+    static void controllerJoystickYCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_AnalogReport* report);
+    static void controllerXAxisCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_AnalogReport* report);
+    static void controllerYAxisCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_AnalogReport* report);
```

Comment:
yep, I left controllerYAxisCallback & controllerXAxisCallback and removed the Joystick callbacks.